tests/integration/test_charm.py

```diff
@@ -5,19 +5,104 @@
+    resources = {"prometheus-tester-image": "python:slim"}
+    app_name = "prometheus-tester"
+
+    await ops_test.model.deploy(
+        prometheus_tester_charm, resources=resources, application_name=app_name
+    )
+    await ops_test.model.wait_for_idle(apps=[app_name], status="active")
+    await ops_test.model.block_until(lambda: len(ops_test.model.applications[app_name].units) > 0)
+
+    assert ops_test.model.applications[app_name].units[0].workload_status == "active"
+
+    await ops_test.model.applications[app_name].remove()
+    await ops_test.model.reset()
+
+
+@pytest.mark.abort_on_fail
+async def test_prometheus_scrape_relation_with_prometheus_tester(
+    ops_test, prometheus_charm, prometheus_tester_charm
+):
+    """Test basic functionality of prometheus_scrape relation interface."""
+    tester_resources = {"prometheus-tester-image": "python:slim"}
+    prometheus_resources = {"prometheus-image": "ubuntu/prometheus:latest"}
+    prometheus_app_name = "prometheus"
+    tester_app_name = "prometheus-tester"
+
+    await ops_test.model.deploy(
+        prometheus_charm, resources=prometheus_resources, application_name=prometheus_app_name
+    )
     await ops_test.model.deploy(
-        prometheus_charm, resources=resources, application_name="prometheus"
+        prometheus_tester_charm, resources=tester_resources, application_name=tester_app_name
     )
-    await ops_test.model.wait_for_idle(apps=["prometheus"], status="active")
+    await ops_test.model.wait_for_idle(apps=[prometheus_app_name], status="active")
+    await ops_test.model.wait_for_idle(apps=[tester_app_name], status="active")
+    await ops_test.model.block_until(
+        lambda: len(ops_test.model.applications[prometheus_app_name].units) > 0
+    )
+    await ops_test.model.block_until(
+        lambda: len(ops_test.model.applications[tester_app_name].units) > 0
+    )
+
+    assert ops_test.model.applications[prometheus_app_name].units[0].workload_status == "active"
+    assert ops_test.model.applications[tester_app_name].units[0].workload_status == "active"
+
+    await check_prometheus_is_ready(ops_test, prometheus_app_name, 0)
+    initial_config = await get_prometheus_config(ops_test, prometheus_app_name, 0)
+    initial_rules = await get_prometheus_rules(ops_test, prometheus_app_name, 0)
+
+    await ops_test.model.add_relation(prometheus_app_name, tester_app_name)
+    await ops_test.model.wait_for_idle(apps=[prometheus_app_name], status="active")
+    config_with_relation = await get_prometheus_config(ops_test, prometheus_app_name, 0)
+    tester_job = get_job_config_for(tester_app_name, config_with_relation)
+    assert tester_job != {}
+
+    rules_with_relation = await get_prometheus_rules(ops_test, prometheus_app_name, 0)
+    tester_rules = get_rules_for(tester_app_name, rules_with_relation)
+    assert tester_rules != {}
+
+    await ops_test.model.applications[tester_app_name].remove()
```

Comment:
I would move this into a different test.
In pytest, tests are run sequentially and the ops_test plugin takes care of cleanup.

Reply:
As mentioned above. I would like each test to be independent of the other and want each test to start with a clean slate.